bscript/interpreter/engine.go

```diff
@@ -22,13 +22,29 @@ func NewEngine() Engine {
 
 // Execute will execute all scripts in the script engine and return either nil
 // for successful validation or an error if one occurred.
+//
+// Execute with tx example:
+//  if err := engine.Execute(interpreter.ExecutionParams{
```

Comment:
Could this be better using something like this `engine.Execute(WithTx(tx, previousOutput,...))` and `engine.Execute(WithScripts(lockingScript, unlockingScript,...))`.

My thought is with the single struct as shown, could be confusing and a user may see all options and think they may all be required ie tx AND script.

I would propose the With func would not be variadic but a single function accepted by the executor, so they also don't do WithTx AND WithScript.

Reply:
Aw good idea